models/members.js

```diff
@@ -99,15 +99,25 @@ const fetchUsersWithRole = async (role) => {
  * @return { Object }: whether moveToMember was successful or not and whether user is already a member or not
  */
 
-const addArchiveRoleToMembers = async (userId) => {
+const addArchiveRoleToMembers = async (userId, currentUserId, reason) => {
   try {
     const userDoc = await userModel.doc(userId).get();
     const user = userDoc.data();
     if (user?.roles && user.roles[ROLES.ARCHIVED]) return { isArchived: true };
     const roles = { ...user.roles, [ROLES.ARCHIVED]: true };
-    await userModel.doc(userId).update({
-      roles,
-    });
+    const archivedDetails = {
```

Comment:
here where you set the archivedDetails property. The value of the reason property is being set to the empty string if the user does not provide a reason for archiving the user. However, the reason property is a required property, so the empty string is not a valid value.

Reply:
if the reason property is an empty string, it will throw an error of `400` bad request, reason required